server/src/db-client.ts

```diff
@@ -0,0 +1,20 @@
+import Knex from "knex";
+import type { Knex as KnexType } from "knex";
+import { logger } from "./logger";
+import { getHostInstance, loadDbConfig } from "./config";
+
+export function createDbClient(label: string): KnexType<any, unknown[]> {
+  const client = Knex(loadDbConfig());
+
+  // Add debug-related logging.
+  const pool = client.client.pool;
```

Comment:
Looks like a light Law of Demeter violation here. Do you think we should guard against `client.client` or `client.client.pool` not existing?

Reply:
I think this is ok — Demeter-wise, this module exists black-box special things we do to Knex, and is well covered by tests and, partially, the TypeScript compiler (`client.client` is publicly typed, but as `any`, even though `Knex.Client` is publicly exported as a type, which is 🤷… I added a new commit with a typecast to use it).

I‘d rather have the tests and/or TSC scream at us in an upgrade PR (cool thing: Tarn’s typings include the event names, so we can’t have a wrong sting in there, either!) so we know to find a new approach to logging than have it silently stop logging if something changes.

![Screen Shot 2022-07-29 at 11 20 19 AM](https://user-images.githubusercontent.com/74178/181821051-94fb9301-c989-4bd7-bb3c-49ba0dcc123b.png)

Reply:
Also submitted a bug on Knex for the types: https://github.com/knex/knex/issues/5283